client/engine/engine.go

```diff
@@ -295,11 +312,31 @@ func (e *Engine) handleMessage(message protocols.Message) (ObjectiveChangeEvent,
 		objective, _ = objective.Reject()
 		err = e.store.SetObjective(objective)
 		if err != nil {
-			return ObjectiveChangeEvent{}, err
+			return EngineChangeEvent{}, err
 		}
 
 		allCompleted.CompletedObjectives = append(allCompleted.CompletedObjectives, objective)
 	}
+
+	for _, voucher := range message.Vouchers() {
+
+		c, ok := e.store.GetChannelById(voucher.ChannelId())
+		if !ok {
+			return EngineChangeEvent{}, fmt.Errorf("could not get channel from the store %s", c.Id)
+		}
+		recipient := c.Participants[2]
+		if recipient != *e.store.GetAddress() {
+			return EngineChangeEvent{}, fmt.Errorf("not the recipient in channel %s", c.Id)
+		}
+		// TODO: return the amount we paid?
+		_, err := e.rm.Receive(voucher)
+
+		allCompleted.ReceivedVouchers = append(allCompleted.ReceivedVouchers, voucher)
+		if err != nil {
+			return EngineChangeEvent{}, fmt.Errorf("error accepting payment voucher: %w", err)
+		}
```

Comment:
That's a good point! I think right now in `handleMessage` we [generally stop and return an error](https://github.com/statechannels/go-nitro/blob/1edd6b56f293a599570d05c3c5088549dbae6545/client/engine/engine.go#L275) but we should probably consider handling a mix of valid/invalid messages.

I think it might make sense to tackle this work in a separate scope?

Reply:
https://github.com/statechannels/go-nitro/issues/840